plugins/ltac2/tac2externals.ml

```diff
@@ -82,8 +82,11 @@ let define : type v f. _ -> (v,f) spec -> f -> unit = fun n ar v ->
     match ar with
     | S (V tr) -> tr v
     | S (F _) ->
-        let cl = Tac2ffi.mk_closure (arity_of ar) (interp_spec ar v) in
-        Tac2ffi.of_closure cl
+      let arity = arity_of ar in
+      let v = interp_spec ar v in
+      let v = Tac2ffi.purify_closure arity v in
+      let cl = Tac2ffi.mk_closure arity v in
+      Tac2ffi.of_closure cl
```

Comment:
I wonder if this is the right approach, perhaps we should instead add flags to the `tac`, `ret` and so on functions (i.e., trust users to use the right spec)? This is pretty low-level stuff, so I think having more flexibility is probably best (I don't want to discover that we have a massive performance regression due to that when we bump to 8.20). Does the test-suite have (diverse) enough Ltac2 code to notice potential regressions?

That is probably what I would have done in the first place if `mk_closure` add the bit of documentation you added here in the first place.

Reply:
The bench has enough to notice potential regressions from such things.

I don't think there's much risk here.